hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleWorkflow.java

```diff
@@ -254,64 +296,105 @@ private void handlePlatformTransaction(
                     feeManager,
                     consensusNow);
 
-            // Now that we have a created handle context object and a consensus timestamp, run the appropriate {@code
-            // ConsensusTimeHook} event handlers
-            stakingPeriodTimeHook.process(consensusNow, context);
-            // @future('7836'): update the exchange rate and call from here
+            // Calculate the fee
+            fees = dispatcher.dispatchComputeFees(context);
 
-            // Dispatch the transaction to the handler
-            dispatcher.dispatchHandle(context);
+            // Run all pre-checks
+            final var preCheckResult = runPreChecks(consensusNow, verifier, preHandleResult);
+            if (preCheckResult.status() != SO_FAR_SO_GOOD) {
+                if (preHandleResult.status() == NODE_DUE_DILIGENCE_FAILURE) {
+                    payer = creator.accountId();
+                }
+                final var penaltyFee = new Fees(fees.nodeFee(), fees.networkFee(), 0L);
+                feeAccumulator.charge(payer, penaltyFee);
+                recordBuilder.status(preCheckResult.responseCodeEnum());
```

Comment:
Since we have this logic in place in `getCurrentPreHandleResult` -> `if (preHandleResult.status() == NODE_DUE_DILIGENCE_FAILURE)` can we just return `preHandleResult` instead of calling the `createPenaltyPayment` method? Looking at the code this should result in charging the node for the due diligence error. It's related to https://github.com/hashgraph/hedera-services/issues/6811